include/sparrow/layout/variable_size_binary_array.hpp

```diff
@@ -32,34 +33,43 @@ namespace sparrow
     template <class L>
     class variable_size_binary_reference;
 
-    template <class L, bool is_const>
+    template <class Layout, typename Iterator_types>
```

Comment:
Maybe you could use the iterator_types concept you've introduced in this PR  to constrain the second template parameter?